internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -419,3 +424,53 @@ func (r ContainerAppEnvironmentResource) Update() sdk.ResourceFunc {
 		},
 	}
 }
+
+func consumptionIsExplicitlyDefined(metadata sdk.ResourceMetaData) bool {
+	config := ContainerAppEnvironmentModel{}
+	if err := metadata.Decode(&config); err != nil {
+		return false
+	}
+	for _, v := range config.WorkloadProfiles {
+		if strings.EqualFold(v.Name, string(helpers.WorkloadProfileSkuConsumption)) {
```

Comment:
this is coming from the config which will have been validated by the Schema, so we shouldn't need the `EqualFold` here?

```suggestion
		if v.Name == string(helpers.WorkloadProfileSkuConsumption) {
```